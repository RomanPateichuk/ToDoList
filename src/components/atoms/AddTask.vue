<template>
  <div :class="$style.new">
    <input
      type="text"
      :class="$style.add"
      placeholder="Add a new task"
      v-model="NewTaskText"
      @keydown.enter="addNewTask"
    />
  </div>
</template>
<script>
export default {
  data: function () {
    return {
      NewTaskText: "",
    };
  },
  methods: {
    addNewTask: function () {
      this.$emit("callAddTask", this.NewTaskText);
      this.NewTaskText = "";
    },
  },
};
</script>
<style lang="scss" module>
@import "@/assets/scss/main.scss";

.new {
  @include Task;
  background: $new-task-bg;
  border: 1.5px dashed $task-remove;
  justify-content: center;
  margin: 1.5625rem 0;
  &:focus {
    ::placeholder {
      opacity: 1;
    }
  }
  &:focus-within {
    background: $main;
  }
  :last-child {
    margin-left: 0;
  }
  .add {
    color: $task-remove;
    width: 100%;
    border: none;
    background-color: inherit;
    &::placeholder {
      color: inherit;
      text-align: center;
    }
  }
}
</style>
