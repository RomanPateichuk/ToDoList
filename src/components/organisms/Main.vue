<template>
  <main :class="$style.main">
    <div :class="$style.tasks">
      <Task
        v-for="(item, index) in getFilter"
        :id="item.id"
        :key="index"
        :isChecked="item.checked"
        :TaskValue="item.value"
      />
      <div v-if="get_tasks_active == 0">Новых задач нет</div>
      <AddTask />
    </div>
  </main>
</template>

<script>
import EventBus from "../../event-bus";
import { mapGetters } from "vuex";
export default {
  mounted() {
    EventBus.$on("SetFilter", (FooterBtnValue) => {
      this.SetFilter(FooterBtnValue);
    });

    EventBus.$on("CallDeleteTask", (id) => {
      this.DeleteTask(id);
    });

    EventBus.$on("CallCompleteTask", (id) => {
      this.CompleteTask(id);
    });

    if (localStorage.tasks) {
      this.$store.commit("UpdateTasks");
    }
  },
  watch: {
    tasks: function () {
      localStorage.setItem(
        "tasks",
        JSON.stringify(this.$store.getters.get_tasks_active)
      );
    },
  },
  components: {
    Task: () => import("@/components/molecules/TaskWrapper.vue"),
    AddTask: () => import("@/components/molecules/AddTaskWrapper.vue"),
  },
  computed: {
    ...mapGetters(["getFilter", "get_tasks_active"]),

    tasks() {
      return this.$store.getters.get_tasks_active;
    },
  },

  methods: {
    SetFilter: function (FooterBtnValue) {
      this.$store.commit("setFilter", FooterBtnValue);
    },

    DeleteTask: function (id) {
      this.$store.commit("DeleteTask", id);
    },

    CompleteTask: function (id) {
      this.$store.commit("CompleteTask", id);
    },
  },
};
</script>

<style lang="scss" module>
@import "@/assets/scss/main.scss";
.main {
  display: flex;
  padding: 0 1.875rem 0 1.875rem;
  flex-direction: column;
  @include MediaMainMobile;

  .tasks {
    margin: 1.875rem 0 0 0;
  }
}
</style>
