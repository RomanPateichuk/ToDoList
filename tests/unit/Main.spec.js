import { mount, createLocalVue } from "@vue/test-utils";
import Main from "../../src/components/organisms/Main.vue";
import AddTask from "../../src/components/atoms/AddTask.vue";
import Tasks from "../../src/components/molecules/TaskWrapper.vue"
import Vuex from "vuex";

const localVue = createLocalVue();
localVue.use(Vuex);

const state = {
  
    tasks: [
      { id: "1", value: "Task 1", checked: false },
      { id: "2", value: "Task 2", checked: false },
      { id: "3", value: "Task 3", checked: false },
      { id: "4", value: "Task 4", checked: false },
      { id: "255", value: "Сделать таск по VUE", checked: false },
      { id: "256", value: "Сделать таск по REACT", checked: false },
      { id: "286", value: "Сделать таск по VUE Todo", checked: false },
    ],
  
    filter: "All",
  };



const getters = {
  getTasks: () => [{ id: "1", value: "Task 1", checked: false }],
  getFilter: () => [],
};

const mutations = {
  addTask: jest.fn(),
  updateTasks: jest.fn(),
};

const store = new Vuex.Store({ getters, mutations, state });

describe("Тестирование компонента Main:", () => {
  const wrapper = mount(Main, { store, localVue });

  test("соответствие снимку", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("вызов события callAddTask из дочернего компонента", () => {
    expect(wrapper.findComponent(AddTask).vm.$emit("callAddTask", "value")).toBeTruthy();
  });

  test("проверка вызова мутации addTask с правильным параметром", () => {
    expect(mutations.addTask).toHaveBeenCalledWith(state, "value");
  });

  test("Подключение компонента 'AddTask'", () => {
    expect(wrapper.findComponent(AddTask).exists()).toBe(true);
  });


  test("Подключение компонента 'TaskWrapper'", () => {
    //expect(wrapper.findAllComponents(Tasks)).toHaveLength(7);
  });
});
