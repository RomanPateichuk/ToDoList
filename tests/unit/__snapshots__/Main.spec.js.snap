// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента Main: соответствие снимку 1`] = `
<main>
  <div>
    <!---->
    <div><input type="text" placeholder="Add a new task"></div>
  </div>
</main>
`;
